src/main/java/rainy/database/Storage.java

```diff
@@ -0,0 +1,90 @@
+package rainy.database;
+
+import java.io.FileWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+import java.util.Scanner;
+
+import rainy.rainyexceptions.InvalidIndexException;
+import rainy.rainyexceptions.InvalidMarkAndUnmarkException;
+
+import rainy.tasks.TaskTracker;
+
+/**
+ * Takes in a <code>File</code> object and either reads the file or writes over it entirely.
+ */
+public class Storage {
+
+    /**
+     * Constructs a new <code>Storage</code> object.
+     */
+    public Storage() {
+
+    }
+
+    /**
+     * Takes in a <code>File</code> object and reads the data.
+     * @param newFile                        Represents the file object to be read.
+     * @return                               This method copies the previously established tasks to a newly created <code>TaskTracker</code>
+     *                                       object and returns it to be used by the Rainy chatbot.
+     * @throws InvalidIndexException         Thrown by <code>TaskManager</code> object when user provides a non-existent task number.
+     * @throws InvalidMarkAndUnmarkException Thrown by <code>Task</code> object when user wants to mark a marked tasked or unmark an unmarked task.
+     */
```

Comment:
I like how you use JavaDocs in most of your public methods. It is complete with param, return and even throw